inc/sitemaps/class-author-sitemap-provider.php

```diff
@@ -0,0 +1,262 @@
+	 *
+	 * @return boolean
+	 */
+	public function handles_type( $type ) {
+
+		return $type === 'author';
+	}
+
+	/**
+	 * @param int $max_entries Entries per sitemap.
+	 *
+	 * @return array
+	 */
+	public function get_index_links( $max_entries) {
+
+		global $wpdb;
+
+		$options = WPSEO_Options::get_all();
+
+		if ( $options['disable-author'] || $options['disable_author_sitemap'] ) {
+			return array();
+		}
+
+		$users = get_users( array( 'who' => 'authors' ) );
+		$users = apply_filters( 'wpseo_sitemap_exclude_author', $users ); // TODO document filter. R.
+		$users = wp_list_pluck( $users, 'ID' );
+
+		$count     = count( $users );
+		$max_pages = ( $count > 0 ) ? 1 : 0;
+
+		if ( $count > $max_entries ) {
+			$max_pages = (int) ceil( $count / $max_entries );
+		};
+
+		// Must use custom raw query because WP User Query does not support ordering by usermeta.
+		// Retrieve the newest updated profile timestamp overall.
+		// TODO order by usermeta supported since WP 3.7, update implementation? R.
+		$date_query = "
+			SELECT mt1.meta_value
+			FROM $wpdb->users
+				INNER JOIN $wpdb->usermeta ON ($wpdb->users.ID = $wpdb->usermeta.user_id)
+				INNER JOIN $wpdb->usermeta AS mt1 ON ($wpdb->users.ID = mt1.user_id)
+			WHERE 1=1
+				AND (
+					($wpdb->usermeta.meta_key = %s AND CAST($wpdb->usermeta.meta_value AS CHAR) != '0')
+					AND mt1.meta_key = '_yoast_wpseo_profile_updated'
+				)
+			ORDER BY mt1.meta_value
+		";
+
+		$index = array();
+
+		for ( $i = 0; $i < $max_pages; $i++ ) {
+
+			$count = ( $max_pages > 1 ) ? ( $i + 1 ) : '';
+
+			if ( empty( $count ) || $count == $max_pages ) {
+				$sql = $wpdb->prepare(
+					$date_query . ' ASC LIMIT 1',
+					$wpdb->get_blog_prefix() . 'user_level'
+				);
+				// Retrieve the newest updated profile timestamp by an offset.
+			}
+			else {
+				$sql = $wpdb->prepare(
+					$date_query . ' DESC LIMIT 1 OFFSET %d',
+					$wpdb->get_blog_prefix() . 'user_level',
+					( ( $max_entries * ( $i + 1 ) ) - 1 )
+				);
+			}
+
+			$date = $wpdb->get_var( $sql );
+
+			$index[] = array(
+				'loc'     => wpseo_xml_sitemaps_base_url( 'author-sitemap' . $count . '.xml' ),
+				'lastmod' => '@' . $date, // @ for explicit timestamp format
+			);
+		}
+
+		return $index;
+	}
+
+	/**
+	 * Get set of sitemap link data.
+	 *
+	 * @param string $type         Sitemap type.
+	 * @param int    $max_entries  Entries per sitemap.
+	 * @param int    $current_page Current page of the sitemap.
+	 *
+	 * @return array
+	 */
+	public function get_sitemap_links( $type, $max_entries, $current_page ) {
+
+		$options = WPSEO_Options::get_all();
+
+		$links = array();
+
+		if ( $options['disable-author'] === true || $options['disable_author_sitemap'] === true ) {
+			return $links;
+		}
+
+		$steps  = $max_entries;
+		$offset = ( $current_page > 1 ) ? ( ( $current_page - 1 ) * $max_entries ) : 0;
+
+		// Initial query to fill in missing usermeta with the current timestamp.
+		$users = get_users( array(
```

Comment:
I would put this in a separate method just to keep the responsibilities of this method small.